### ISSUE TITLE
rails_rbi:models fails if a table is created with id: false

### ISSUE THREAD

**Issue by tmcw** [chanzuckerberg/sorbet-rails#373]
**Describe the bug:**

Associations between tables that don't go through an index named precisely `id` fail to generate proper sorbet types when using sorbet-rails. Here's an example error message when this failure occurs:

```
-- Generate sigs for Example --
!!! Plugin SorbetRails::ModelPlugins::ActiveRecordAssoc threw an exception: undefined method `sql_type' for nil:NilClass
```

**Steps to reproduce:**

Create a table relationship that does not use a literal `id` column, or use a gem that relies on such a table relationship, like [`closure_tree`](https://github.com/ClosureTree/closure_tree). I've provided an example repository here:

https://github.com/tmcw/sorbet-rails-example

**Expected behavior:**

Sorbet-rails should be able to generate associations between tables based on specified indices and have no strict requirements about column naming.

**Versions:**

- Ruby: 2.6.6p146
- Rails: 6.0.3.2
- Sorbet:  0.5.5871
- Sorbet-Rails: 0.7.1

---

I suspect that this bug is originating here, where there is an expected id column:

https://github.com/chanzuckerberg/sorbet-rails/blob/83360d1eff256d4c540d1e01f236a0f59da73b7f/lib/sorbet-rails/model_plugins/active_record_assoc.rb#L137-L143

**Comment by manhhung741** [comment]
Thanks for flagging. Is there a different primary key for the table? 
This may be fixed by #369.

**Comment by manhhung741** [comment]
I think this issue is fixed by #369. Please let me know if the issue still persists for you
